src/librbd/io/ObjectRequest.cc

```diff
@@ -521,11 +522,21 @@ void AbstractObjectWriteRequest<I>::handle_write_object(int r) {
   ldout(image_ctx->cct, 20) << "r=" << r << dendl;
 
   r = filter_write_result(r);
-  if (r == -ENOENT || (r == -ERANGE && !image_ctx->migration_info.empty())) {
+  if (r == -ENOENT) {
```

Comment:
@dillaman Just thinking, is there the same potential issue with an ordinary parent gone?
Also, according to the current code if `-ENOENT` is returned but it is not a copyup, the error is just ignored. It looks like it should never happen, but anyway shouldn't we allow the error to bubble up (just in case)?

Reply:
The only way that could happen that I can think of is if the pool is deleted while IO is in-flight. I suppose we can let the ```-ENOENT``` bubble up in the case where ```!m_copyup_enabled```.